dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/AuditLoggerTest.java

```diff
@@ -0,0 +1,356 @@
+        newLogger.setRetryInterval(10);
+        logger.setRetryInterval(1);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("retryInterval is not as expected", logger.getRetryInterval(), is(10));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditLoggerInstalled_WhenCalled() {
+        newLogger.setAuditLoggerInstalled(false);
+        logger.setAuditLoggerInstalled(true);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditLoggerInstalled is not as expected", logger.getAuditLoggerInstalled(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditRecordRepositoryDevices_WhenCalled() {
+        final Connection auditUDP = new Connection("audit-udp", "localhost", 514);
+        auditUDP.setProtocol(Connection.Protocol.SYSLOG_UDP);
+
+        final AuditRecordRepository arr1 = new AuditRecordRepository();
+        arr1.addConnection(auditUDP);
+
+        final Device arrDevice1 = new Device("syslog");
+        arrDevice1.addDeviceExtension(arr1);
+        arrDevice1.addConnection(auditUDP);
+
+        newLogger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice1));
+
+        final Connection auditTCP = new Connection("audit-tcp", "localhost", 516);
+        auditTCP.setProtocol(Connection.Protocol.SYSLOG_TLS);
+
+        final AuditRecordRepository arr2 = new AuditRecordRepository();
+        arr2.addConnection(auditTCP);
+
+        final Device arrDevice2 = new Device("syslog2");
+        arrDevice2.addDeviceExtension(arr2);
+        arrDevice2.addConnection(auditTCP);
+
+        logger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice2));
+
+        logger.reconfigure(newLogger);
+
+        final Device toDevice = logger.getAuditRecordRepositoryDevices().get(0);
+        assertThat("auditRecordRepositoryDevices' device name is not as expected", toDevice.getDeviceName(), is(arrDevice1.getDeviceName()));
+        assertThat("auditRecordRepositoryDevices' device extension is not as expected", toDevice.getDeviceExtension(AuditRecordRepository.class), samePropertyValuesAs(arr1));
+        assertThat("auditRecordRepositoryDevices' connections is not as expected", toDevice.getConnections().get(0), samePropertyValuesAs(auditUDP));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditSuppressCriteriaList_WhenCalled() {
+        final AuditSuppressCriteria newCriteria = new AuditSuppressCriteria();
+        newCriteria.setCommonName("NewCommonName");
+        newLogger.setAuditSuppressCriteriaList(Collections.singletonList(newCriteria));
+
+        final AuditSuppressCriteria oldCriteria = new AuditSuppressCriteria();
+        oldCriteria.setCommonName("OldCommentName");
+        logger.setAuditSuppressCriteriaList(Collections.singletonList(oldCriteria));
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditSuppressCriteriaList is not as expected", logger.getAuditSuppressCriteriaList().get(0), samePropertyValuesAs(newCriteria));

```

Comment:
Not sure why you are using here this matcher.

" Creates a matcher that matches when the examined object has values for all of its JavaBean properties that are equal to the corresponding values of the specified bean."

http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/beans/SamePropertyValuesAs.html

This will compare fields, which is unrelated to the AuditLogger class, which simply clears the list and adds new filters. You should be able to use equalTo() and compare the list, which will iterate through list and do a object reference comparison.

Reply:
I'll change to compare the AuditSuppressCriteria.